.github/workflows/paths-ignore.yaml

```diff
@@ -0,0 +1,80 @@
+# This is a reusable workflow to determine if the current change requires an E2E
+# run.  This is required because using [paths-ignored] directly means the whole
+# workflow is skipped, but that means that it doesn't count as having run a
+# required workflow.
+
+# Usage:
+#   jobs:
+#     check_paths:
+#        uses: ./.github/workflows/actions/paths-ignore.yaml
+#     do_thing:
+#        if: jobs.check_paths.outputs.should_run == 'true'
+#        # Unfortunately, a string comparison is required.
+
+name: Check for ignored paths
+
+on:
+  workflow_call:
+    inputs:
+      paths_ignore:
```

Comment:
I think the input/output parameters should use hyphens instead of underscores: `paths-ignore` and `should-run`.

Especially `path-ignore` would match the same key used for triggers, and it seems to be the general style of GitHub actions to use hyphenated names, e.g.

```yaml
    - uses: actions/setup-go@v5
      with:
        go-version: '^1.21'
        cache-dependency-path: src/go/**/go.sum
```

There are some weird exceptions, like the trigger names themselves (`pull_request`, `workflow_dispatch`), but the vast number of names use hyphens.